src/RestServer/Controllers/WalletController.cs

```diff
@@ -0,0 +1,672 @@
+// Copyright (C) 2015-2023 The Neo Project.
+//
+// The Neo.Plugins.RestServer is free software distributed under the MIT software license,
+// see the accompanying file LICENSE in the main directory of the
+// project or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Akka.Actor;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Neo.Network.P2P.Payloads;
+using Neo.Plugins.RestServer.Exceptions;
+using Neo.Plugins.RestServer.Helpers;
+using Neo.Plugins.RestServer.Models.Error;
+using Neo.Plugins.RestServer.Models.Wallet;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.Wallets;
+using Neo.Wallets.NEP6;
+using System.Net.Mime;
+using System.Numerics;
+
+namespace Neo.Plugins.RestServer.Controllers
+{
+    /// <summary>
+    /// Wallet API
+    /// </summary>
+    [Route("/api/v1/wallet")]
+    [DisableCors]
+    [Produces(MediaTypeNames.Application.Json)]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [ApiExplorerSettings(GroupName = "v1")]
+    [ApiController]
+    public class WalletController : ControllerBase
+    {
+        internal static WalletSessionManager WalletSessions { get; } = new();
+
+        private readonly NeoSystem _neosystem;
+        private readonly RestServerSettings _settings;
+
+        /// <summary>
+        /// CTOR
+        /// </summary>
+        /// <exception cref="NodeNetworkException">Node network doesn't match plugins network.</exception>
+        public WalletController()
+        {
+            _neosystem = RestServerPlugin.NeoSystem ?? throw new NodeNetworkException();
+            _settings = RestServerSettings.Current;
+        }
+
+        /// <summary>
+        /// Opens a wallet.
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns>A newly created wallet session object.</returns>
+        /// <response code="200">Returns newly create wallet session object.</response>
+        /// <response code="400">If anything is invalid or request crashes.</response>
+        [HttpPost("open", Name = "WalletOpen")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(WalletSessionModel))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ErrorModel))]
+        public IActionResult WalletOpen(
```

Comment:
If the wallet file is open and another request comes in to open the same wallet file. Should we create a new session (_this is currently what it does_) or return the already opened wallet's session? I don't know if the Neo wallet API will care. For example if one was to change the password of the wallet and the wallets are already open. Does that have any effect on the wallet for making or transferring funds.